src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java

```diff
@@ -194,6 +194,74 @@
 
     private String minimalApiVersion;
 
+    protected String getAutoPull() {
+        return getProperty("autoPull");
```

Comment:
What's about the field which would be set by Maven from the XML configuration ? I don't see where these are used in the getters. These should be actually the fallback values when no property is specified and should mimic the same algorithm that a `@Parameter(property=...)` would do. E.g. I'm not 100% sure what has precedence, the XML value, a system property and a Maven property.

Reply:
Sorry, I didn't consider XML while testing

Reply:
I think this should have been caught by integrations tests, but I don't think we have any ;-)